controllers/logging/logging_controller.go

```diff
@@ -93,6 +93,28 @@ func (r *LoggingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
+	allLoggings := &loggingv1beta1.LoggingList{}
+	if err := r.Client.List(ctx, allLoggings); err != nil {
+		return reconcile.Result{}, errors.WrapIf(err, "listing logging resources")
+	}
+
+	loggingsForTheSameRef := make([]string, 0)
+	for _, l := range allLoggings.Items {
+		if l.Name == logging.Name {
+			continue
+		}
+		if l.Spec.LoggingRef == logging.Spec.LoggingRef {
+			loggingsForTheSameRef = append(loggingsForTheSameRef, l.Name)
+		}
+	}
+
+	if len(loggingsForTheSameRef) > 0 {
+		problem := fmt.Sprintf("multiple other logging resources exist with the same loggingRef: %s",
+			strings.Join(loggingsForTheSameRef, ","))
+		logging.Status.Problems = []string{problem}
```

Comment:
```suggestion
		logging.Status.Problems = append(logging.Status.Problems, problem)
```
this should be less error prone, but we should also clear the problem list since we're going to rebuild it as part of reconciliation

Reply:
actually I wanted to update the logging resource right here since we return from the reconciliation right after this line, but I forgot...

Reply:
completely restructure the code